com.minres.coredsl/src/com/minres/coredsl/CoreDsl.xtext

```diff
@@ -342,49 +342,32 @@ PostfixExpression:
 	|	{PostfixExpression.left=current} op='--'
 	)*;
 
-PrimaryExpression
-    :	ref=[Variable]
-    |   constant=Constant
-    |   literal+=StringLiteral+
-    |   '(' left=ConditionalExpression ')'
-    ;
+PrimaryExpression: IdentifierReference | Constant | ParenthesisExpression;
 
-Variable hidden()
-	:	FunctionDefinition|DirectDeclarator|Field
-	;
+ParenthesisExpression: '(' inner=ConditionalExpression ')';
 
-StringLiteral
-    :   value=ENCSTRINGCONST
-	|   value=STRING
-    ;
+IdentifierReference: identifier=[Identifier];
 
-ConstantExpression returns Expression
-    :   ConditionalExpression
-    ;
+Identifier hidden(): FunctionDefinition | DirectDeclarator | BitField;
```

Comment:
Identifier is not the right naming as it is not an identifier rather than something identifiable. Variable is not the right naming either.

Reply:
I guess the correct term would be 'declaration', but that is already used. Any suggestions for a better name?

Reply:
Shouldn't `IdentifierReference` just point to an `ID` terminal, syntactically?

Reply:
`IdentifierReference: identifier=[Identifier];` means an ID belonging to an Identifier but XText derives a reference (based on an ID and scoping rules).

@AtomCrafty It is not an declaration since it denotes something addressable or locatable (other than e.g. a constant). So `Adressable`, `Location` or `Entity` might be a possible naming in my opinion

Reply:
`NamedEntity` SGTM.

Reply:
Sounds reasonable. Fine with me.

Reply:
Can we merge this as-is and I'll rename it in a separate PR? Because otherwise I'll have to resolve a lot of conflicts with the other PRs.